task/engine/src/main/java/io/knotx/fragments/task/engine/NodeFatalException.java

```diff
@@ -12,23 +12,21 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
- *
- * The code comes from https://github.com/tomaszmichalak/vertx-rx-map-reduce.
  */
-package io.knotx.fragments.task.engine.exception;
-
-import io.knotx.fragments.api.Fragment;
+package io.knotx.fragments.task.engine;
 
 public class NodeFatalException extends IllegalStateException {
 
-  private Fragment fragment;
+  public NodeFatalException(String message) {
+    super(message);
+  }
 
-  public NodeFatalException(Fragment fragment) {
-    super("Failed during fragment processing [" + fragment.getId() + "]");
-    this.fragment = fragment;
+  public NodeFatalException(Throwable cause) {
+    super(cause);
   }
 
-  public Fragment getFragment() {
-    return fragment;
+  public NodeFatalException(String message, Throwable cause) {
```

Comment:
As above

Reply:
This class is moved to Task API. Each action can throw an NodeFatalException to prevent further task processing.